### PR TITLE
Splitted RANS3PF module

### PR DESCRIPTION
**Pull request by JohanMabille**
# Mandatory Checklist

Please ensure that the following criteria are met:

- [x] Title of pull request describes the changes/features
- [x] Request at least 2 reviewers
- [x] If new files are being added, the files are no larger than 100kB. Post the file sizes.
- [x] Code coverage did not decrease. If this is a bug fix, a test should cover that bug fix. If a new feature is added, a test should be made to cover that feature.
- [x] New features have appropriate documentation strings (readable by sphinx)
- [x] Contributor has read and agreed with CONTRIBUTING.md and has added themselves to CONTRIBUTORS.md 

As a general rule of thumb, try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

# Description

### CODE REVIEW

**Comment by JohanMabille** [comment]
The split of RANS3PF fixes the memory issue but now we the file takes too long to compile, resulting in travis time out ...

**Comment by cekees** [comment]
> The split of RANS3PF fixes the memory issue but now the file takes too long to compile, resulting in travis time out ...

To be clear: the memory issue we are seeing is on Docker Hub, not travis, and it's still there on your branch: https://hub.docker.com/repository/registry-1.docker.io/erdc/proteus/builds/cc7c9588-ab4a-4ff2-a56a-bf391be73638

I'm guessing you can fix the compile time issue by modifying the Travis file to use PROTEUS_OPT='-g0 -O0 -DNEBUG'' as I mentioned on the proteus channel.

**Comment by JohanMabille** [comment]
I think we're talking about two different memory issues, sorry for not being clear. Without the split of RANS3PF module, we had a memory issue when trying to build with -O2 or -O3 on travis (only on hdist). Thus the switch to -O0 in #1128. Now, with the split, we don't have this memory issue anymore, but the build takes too long.

**Comment by cekees** [comment]
I see you split RANS3PF first, so you may see more of an improvement when you split RANS3PSed, which seems to be the most resource-intensive compile. I think we should go ahead and split RANS2P as well for consistency (probably has marginal effect on compile times).

**Comment by JohanMabille** [comment]
I think the `RANS2P` module has already been splitted.

**Comment by JohanMabille** [comment]
@cekees @zhang-alvin @tridelat I think is one is ready to be merge.

File: .gitignore [hunk 1]
```diff
@@ -62,6 +62,7 @@ proteus/equivalent_polynomials.cpp
 proteus/fenton/Fenton.cpp
 proteus/flcbdfWrappers.cpp
 proteus/mbd/CouplingFSI.cpp
+proteus/MeshAdaptPUMI/Adapter.cpp
```
Comment (zhang-alvin) [inline .gitignore:65]: @JohanMabille Thanks for adding this.

**Comment by JohanMabille** [comment]
> So to clarify, the "split" referred to in this PR is separating the 2D RANS3PF module from the 3D RANS3PF module at compile time? Likewise for RANS3PFSed.

Yes.

**Pull request merged**